### PR TITLE
Add More producer knobs

### PR DESCRIPTION
**Pull request by srkukarni**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation
This pr adds more Producer Config knobs to FunctionConfig and SourceConfig.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jerrypeng** [comment]
@srkukarni why are we generating the new protobuf code for python when we are not adding the support for the new configs in python functions?

**Comment by jerrypeng** [comment]
@srkukarni if we are going to add configs that aren't support in Python or GO, we should either return a error to the user if a user tries to set one of these configs or document it well some where

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java [hunk 1]
```diff
@@ -110,6 +108,22 @@ protected PulsarSinkProcessorBase(Schema schema) {
                 if (pulsarSinkConfig.getProducerSpec().getMaxPendingMessagesAcrossPartitions() != 0) {
                     builder.maxPendingMessagesAcrossPartitions(pulsarSinkConfig.getProducerSpec().getMaxPendingMessagesAcrossPartitions());
                 }
+                if (!pulsarSinkConfig.getProducerSpec().getDisableBatching()) {
+                    builder.enableBatching(true);
```
Comment (sijie) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java:112]: The batching is enabled by default in the builder. So I don't think this change is able to disable batching.

File: pulsar-common/src/main/java/org/apache/pulsar/common/functions/ProducerConfig.java [hunk 2]
```diff
@@ -36,4 +36,8 @@
     private Integer maxPendingMessages;
     private Integer maxPendingMessagesAcrossPartitions;
     private Boolean useThreadLocalProducers;
+    private Boolean disableBatching;
+    private Integer batchingMaxPublishDelay;
```
Comment (sijie) [inline pulsar-common/src/main/java/org/apache/pulsar/common/functions/ProducerConfig.java:40]:
```suggestion
    private Integer batchingMaxPublishDelayMs;
```

**Comment by codelipenghui** [comment]
@srkukarni Could you please take a look at the last comment?

**Comment by codelipenghui** [comment]
move to 2.8.0

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

**Pull request closed without merging**